package/src/package/manifest.rs

```diff
@@ -15,30 +15,34 @@ use crate::Dependency;
 #[cfg_attr(feature = "arbitrary", derive(proptest_derive::Arbitrary))]
 pub struct PackageManifest {
     /// The set of exports in this package.
-    pub exports: BTreeSet<PackageExport>,
+    pub exports: BTreeMap<QualifiedProcedureName, PackageExport>,
     /// The libraries (packages) linked against by this package, which must be provided when
     /// executing the program.
     pub dependencies: Vec<Dependency>,
 }
 
 /// A procedure exported by a package, along with its digest and signature (will be added after
 /// MASM type attributes are implemented).
-#[derive(Clone, PartialEq, Eq, Ord, PartialOrd)]
+#[derive(Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "arbitrary", derive(proptest_derive::Arbitrary))]
 pub struct PackageExport {
     /// The fully-qualified name of the procedure exported by this package
     pub name: QualifiedProcedureName,
     /// The digest of the procedure exported by this package
     #[cfg_attr(feature = "arbitrary", proptest(value = "Word::default()"))]
     pub digest: Word,
+    /// The type signature of the exported procedure
+    #[cfg_attr(feature = "arbitrary", proptest(value = "None"))]
+    pub signature: Option<FunctionType>,
 }
 
 impl fmt::Debug for PackageExport {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let Self { name, digest } = self;
+        let Self { name, digest, signature } = self;
```

Comment:
we'd need a `#[cfg_attr(feature = "arbitrary", proptest(value = "None"))]` for the `signature` field here right?

Reply:
Remember it's `#[cfg_attr]`, not `#[cfg]`, i.e. what is conditionally enabled is the `#[proptest(value = "None")]` attribute, not the `signature` field itself.